lib/types.js

```diff
@@ -477,7 +477,7 @@ function extractUnionInfo(ctx, idlType, errPrefix) {
       seen.object = true;
     } else if (item.idlType === "boolean") {
       seen.boolean = item;
-    } else if (item.idlType === "Function") {
+    } else if (item.idlType === "Function" || ctx.callbackInterfaces.has(item.idlType)) {
```

Comment:
This should be reverted. Callback interfaces are handled below. (Also, callback interfaces and callback functions have different rules for distinguishability.)